File: v131016/TeeNew/BackWall_Image.pas
unit BackWall_Image;
{$I TeeDefs.inc}

interface

uses
  {$IFNDEF LINUX}
  Windows, Messages,
  {$ENDIF}
  SysUtils, Classes,
  {$IFDEF CLX}
  QGraphics, QControls, QForms, QDialogs, QExtCtrls, QStdCtrls, QComCtrls,
  {$ELSE}
  Graphics, Controls, Forms, Dialogs, ExtCtrls, StdCtrls, ComCtrls,
  {$ENDIF}
  {$IFNDEF LINUX}
  {$IFNDEF CLR}
  Jpeg,
  {$ENDIF}
  {$ENDIF}
  Base, TeeProcs, TeEngine, Chart;

type
  TBackWallImage = class(TBaseForm)
    CheckBox1: TCheckBox;
    Image1: TImage;
    Button1: TButton;
    procedure CheckBox1Click(Sender: TObject);
    procedure FormCreate(Sender: TObject);
    procedure Button1Click(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

implementation

{$IFNDEF CLX}
{$R *.DFM}
{$ELSE}
{$R *.xfm}
{$ENDIF}

Uses TeeBrushDlg;

procedure TBackWallImage.CheckBox1Click(Sender: TObject);
begin
  inherited;
  Chart1.BackWall.Transparent:=not CheckBox1.Checked
end;

procedure TBackWallImage.FormCreate(Sender: TObject);
begin
  inherited;
  Chart1.BackWall.Transparent:=False;
  Chart1.BackWall.Brush.Image.Assign(Image1.Picture);

  Chart1.BackWall.Brush.Color:=clGreen;
end;

procedure TBackWallImage.Button1Click(Sender: TObject);
begin
  EditChartBrush(Self,Chart1.BackWall.Brush)
end;

initialization
  RegisterClass(TBackWallImage);
end.

File: TeeNew/Builder2010/BackWall_Image.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "BackWall_Image.h"
#include <TeeBrushDlg.hpp>
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "Base"
#pragma resource "*.dfm"
TBackWallImage *BackWallImage;
//---------------------------------------------------------------------------
__fastcall TBackWallImage::TBackWallImage(TComponent* Owner)
        : TBaseForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TBackWallImage::FormCreate(TObject *Sender)
{
  Chart1->BackWall->Transparent = false;
  Chart1->BackWall->Brush->Image->Assign(Image1->Picture);
  Chart1->BackWall->Brush->Color = clGreen;
}
//---------------------------------------------------------------------------

void __fastcall TBackWallImage::CheckBox1Click(TObject *Sender)
{
  Chart1->BackWall->Transparent = (! CheckBox1->Checked);
}
//---------------------------------------------------------------------------

void __fastcall TBackWallImage::Button1Click(TObject *Sender)
{
  EditChartBrush(this,Chart1->BackWall->Brush);
}
//---------------------------------------------------------------------------

